File: Leetcode/python/twosumv2.py
from typing import List

class Solution :
    #function to find the indices of the two elements that add up to the target
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap ={}
        
        #iterate through each element in the list
        for i,n in enumerate(nums):
            #calculate the difference between the target and the current element
            diff = target - n
            #check if the difference is in the hashmap
            if diff in hashmap:
                #return the indices of the two elements
                return [hashmap[diff],i]
            #add the current element to the hashmap
            hashmap[n] = i

File: Leetcode/javascript/twosumv2.js
const twoSum = (nums, target) => {
    // Create a new Map to store the numbers and their indices
    const map = new Map();
    
    // Iterate through the array of numbers
    for (let index = 0; index < nums.length; index++) {
        let num = nums[index];
        // Calculate the complement of the current number
        let element = target - num;
        
        // Check if the complement exists in the map
        if (map.has(element)) {
            // If it exists, return the indices of the complement and the current number
            return [map.get(element), index];
        }
        
        // If it doesn't exist, add the current number and its index to the map
        map.set(nums[index], index);
    }
    
    // If no solution is found, return an empty array
    return [];
};